### ISSUE TITLE
Unexpected token '=' in Strapi internal middleware module

### ISSUE THREAD

**Issue by khoanguyen-hawking** [strapi/strapi#10528]
## Bug report

Got unexpected error `Unexpected token '='`  in Strapi internal middleware module

### Describe the bug

Full error logs

```
[2021-06-22T22:38:54.739Z] error SyntaxError: Unexpected token '='
    at Function (<anonymous>)
    at /app/node_modules/lodash/lodash.js:14942:16
    at apply (/app/node_modules/lodash/lodash.js:487:27)
    at /app/node_modules/lodash/lodash.js:15326:16
    at apply (/app/node_modules/lodash/lodash.js:489:27)
    at /app/node_modules/lodash/lodash.js:6627:16
    at Function.template (/app/node_modules/lodash/lodash.js:14941:20)
    at serveIndexPage (/app/node_modules/strapi/lib/middlewares/public/index.js:52:29)
    at async /app/node_modules/strapi/lib/middlewares/parser/index.js:48:23
    at async /app/node_modules/strapi-plugin-sentry/middlewares/sentry/index.js:13:9
    at async /app/node_modules/strapi/lib/middlewares/xss/index.js:26:9
[2021-06-22T22:38:54.739Z] debug GET index.html (44 ms) 500
```

### Steps to reproduce the behavior

Access the index page url

### Expected behavior

Get rid of this error because the Sentry plugin will continuously capture it

### System

- Node.js version: v14.17.1
- NPM version:
- Strapi version: 3.6.3 (Community Edition)
- Database: Postgres
- Operating system: Ubuntu 18.04.5 LTS

### Additional context

Environment
![image](https://user-images.githubusercontent.com/84629610/123026451-65569f00-d406-11eb-85b5-5ed74ad6b8bb.png)

Sentry issue: https://sentry.io/share/issue/c5ca717eb52b480ca0ac43313e6e8c3b/

**Comment by kasonde** [comment]
Hi, @khoanguyen-hawking did this issue occur after a specific significant action i.e a version upgrade, a plugin install? customization or something similar?

**Comment by khoanguyen-hawking** [comment]
> Hi, @khoanguyen-hawking did this issue occur after a specific significant action i.e a version upgrade, a plugin install? customization or something similar?

Hi @kasonde,

- No version upgrade.

- Not sure about the customization because I customized a lot, ranging from users&permissions, content-manager plugins as well as a little bit customization on the admin panel. Additionally, after checking the source code (Strapi internal modules) from which the error occurs, I think the reason not coming from our customization.

- Regarding the plugin install, I am not so sure about it. But it is only when I integrated `strapi-plugin-sentry` do I notice starting noticing this error because it is captured by Sentry. Actually I am not sure it exists before the sentry installation or not.

Hope these help! Thanks

**Comment by derrickmehaffy** [comment]
Tested on v3.6.5 with a fresh self-hosted Sentry instance and could not reproduce the issue.

**Comment by derrickmehaffy** [comment]
@khoanguyen-hawking We will need to rule out your customizations as we cannot provide support for those as we didn't write the code. If you can please see if you can reproduce this on a fresh application, it will be needed for us to see if the issue is in our code-base.

**Comment by khoanguyen-hawking** [comment]
Hi @derrickmehaffy, @kasonde

Sorry for late response. Here is an update for it!

It eventually turns out the issue is coming from our own customization. We changed the `lodash.template` settings for our email templating purposes and it conflicts with the `lodash.template` Strapi uses internally (which use the default settings of `lodash.template`.

We initially had this
```
_.templateSettings.interpolate = /{{([\s\S]+?)}}/g;
const _ = require('lodash');

module.exports = {
  template(template, data) {
    return _.template(template)(data);
  },
};
```
which changes the `lodash.template` global setting.

If we change setting the `lodash.template` as below, no more error:
```
const _ = require('lodash');

module.exports = {
  template(template, data) {
    return _.template(template, { interpolate: /{{(.+?)}}/g })(data);
  },
};
```

Sorry for wasting your time and really appreciate your support.
Closing the issue now. Thanks.
